test/files/run/enums.scala

```diff
@@ -118,7 +118,8 @@ object SerializationTest {
   }
 
   def run: Unit = {
-    serialize(new B())
+    /* This is no longer possible with the proxy-based serialization for collections: */
+    //serialize(new B())
```

Comment:
I've given up playing whack-a-mole with Enumeration. If anyone else has an idea, have a go at it. The cyclic dependency problem with deserialized collections is not entirely new. `List` already used the serialization proxy pattern, so any `List`-based code was affected. Now we have more instances because most other collection types (in particular for this case, the default `Set` implementation) do the same.